src/main/java/Duke.java

```
@@ -1,10 +1,273 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import java.nio.file.FileAlreadyExistsException;
+
 public class Duke {
+    public static ArrayList<Task> entireList = new ArrayList<>();
+    public static int counterList = 0;
+    public static boolean isExit = false;
+    public static String filePath = "docs/duke.txt";
+    public static String directoryPath = "docs";
+
+    //Constants
+    static final int LENGTH_OF_BY = 3; // or LENGTH_OF_AT
+    static final int LENGTH_OF_DONE = 4;
+    static final int LENGTH_OF_DELETE = 6;
+    static final int LENGTH_OF_TODO = 4;
+    static final int LENGTH_OF_EVENT = 5;
+    static final int LENGTH_OF_DEADLINE = 8;
+    static final int LENGTH_OF_INPUT_FORMAT = 6; //[✘][✓]
+    static final int LENGTH_OF_INPUT_DONE_STATUS = 4;
+    static final int TICK_HTML_CODE = 10003;
+
+    public static void fileReading (String filePath) throws IOException {
+        try {
+            Files.createDirectories(Paths.get(directoryPath));
+            Files.createFile(Path.of(filePath));
+
+        } catch (FileAlreadyExistsException e) {
+            //Ignore
+        }
```

We should avoid empty catch blocks. Following the recommendation on the code quality section of the textbook, perhaps (if you really can't figure out how to process the catch) you could give a comment to explain why the catch block is left empty.